src/foam/nanos/approval/ApprovableAwareDAO.js

```diff
@@ -0,0 +1,383 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.approval',
+  name: 'ApprovableAwareDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  javaImports: [
+    'foam.comics.v2.userfeedback.UserFeedback',
+    'foam.comics.v2.userfeedback.UserFeedbackAware',
+    'foam.comics.v2.userfeedback.UserFeedbackException',
+    'foam.comics.v2.userfeedback.UserFeedbackStatus',
+    'foam.core.Detachable',
+    'foam.core.FObject',
+    'foam.core.PropertyInfo',
+    'foam.core.X',
+    'foam.dao.AbstractSink',
+    'foam.dao.ArraySink',
+    'foam.dao.DAO',
+    'foam.lib.PropertyPredicate',
+    'foam.mlang.MLang',
+    'foam.mlang.MLang.*',
+    'foam.mlang.predicate.Predicate',
+    'foam.nanos.approval.Approvable',
+    'foam.nanos.approval.ApprovableAware',
+    'foam.nanos.approval.ApprovalRequest',
+    'foam.nanos.approval.ApprovalStatus',
+    'foam.nanos.approval.RoleApprovalRequest',
+    'foam.nanos.auth.LifecycleAware',
+    'foam.nanos.auth.LifecycleState',
+    'foam.nanos.auth.User',
+    'foam.nanos.auth.UserQueryService',
+    'foam.nanos.logger.Logger',
+    'foam.nanos.ruler.Operations',
+    'foam.util.SafetyUtil',
+    'java.util.ArrayList',
+    'java.util.Iterator',
+    'java.util.List',
+    'java.util.Map',
+    'java.util.Set',
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'daoKey'
+    },
+    {
+      class: 'Class',
+      name: 'of'
+    },
+    {
+      class: 'Boolean',
+      name: 'isEnabled',
+      value: true
```

Comment:
Alternative, could this be a rule?

Reply:
Yeah I think we could make it a rule now that I think about it, we previously prevented it because we need a case where it would return null on CREATE and null on REMOVE. But on creates there will always be a new object and on removes if its using lifecycleState it will never hit remove_ since it is actually a put_.

Reply:
Future PR - after merge.